src/pip/_internal/commands/show.py

```diff
@@ -140,41 +141,103 @@ def get_requiring_packages(package_name):
         yield package
 
 
-def print_results(distributions, list_files=False, verbose=False):
+def get_package_info(distributions, list_files=False, verbose=False):
     """
-    Print the information from installed distributions found.
+    Build a dictionary with  information from installed distributions found.
     """
-    results_printed = False
+    pkg_infos_list = []
+
     for i, dist in enumerate(distributions):
-        results_printed = True
-        if i > 0:
-            write_output("---")
 
-        write_output("Name: %s", dist.get('name', ''))
-        write_output("Version: %s", dist.get('version', ''))
-        write_output("Summary: %s", dist.get('summary', ''))
-        write_output("Home-page: %s", dist.get('home-page', ''))
-        write_output("Author: %s", dist.get('author', ''))
-        write_output("Author-email: %s", dist.get('author-email', ''))
-        write_output("License: %s", dist.get('license', ''))
-        write_output("Location: %s", dist.get('location', ''))
-        write_output("Requires: %s", ', '.join(dist.get('requires', [])))
-        write_output("Required-by: %s", ', '.join(dist.get('required_by', [])))
+        pkg_info = {}
+
+        pkg_info["Name"] = dist.get('name', '')
+        pkg_info["Version"] = dist.get('version', '')
+        pkg_info["Summary"] = dist.get('summary', '')
+        pkg_info["Home-page"] = dist.get('home-page', '')
+        pkg_info["Author"] = dist.get('author', '')
+        pkg_info["Author-email"] = dist.get('author-email', '')
+        pkg_info["License"] = dist.get('license', '')
+        pkg_info["Location"] = dist.get('location', '')
+        pkg_info["Requires"] = ', '.join(dist.get('requires', []))
```

Comment:
For the json formatting, this could stay as a list (same for `pkg_info["Required-by"]`) so the `', '.join` could happen in `print_results_default`

Reply:
Hi @xavfernandez 

I have made the changes accordingly. Please take a look.